### PR TITLE
Add tslint rule to use single quotes

### PR DESCRIPTION
**Pull request by thegecko**
Signed-off-by: Rob Moran <rob.moran@arm.com>

As discussed on gitter, the guidelines for strings in the [coding guidelines](https://github.com/theia-ide/theia/wiki/Coding-Guidelines) is confusing and cannot be checked/enforced with linting.

The proposal is to move to using single-quotes everywhere and highlight this using a `tslint` rule.

If agreeable, the coding guidelines can be updated and this PR merged to add warnings for double-quotes.

### CODE REVIEW

File: configs/warnings.tslint.json [hunk 1]
```diff
@@ -47,6 +47,15 @@
...
       "severity": "warning",
       "options": true
+    },
+    "quotemark": {
+      "severity": "warning",
+      "options": [
+        true,
+        "single",
+        "jsx-single",
+        "avoid-escape"
```
Comment (simark) [inline configs/warnings.tslint.json]: Should we use `"avoid-template"` too?

Comment (thegecko) [reply configs/warnings.tslint.json]: Can do, what are people's thoughts?

**Comment by thegecko** [comment]
I just used the github tool to resolve conflicts, looks like it decided to merge rather than rebase

**Comment by thegecko** [comment]
I'll re-create this PR using rebase

**Comment by simark** [comment]
FYI, you could have re-used the same PR by doing the rebase locally, and then pushing with `-f` to the branch.

**Pull request closed without merging**